kimchi/src/circuits/constraints.rs

```diff
@@ -153,6 +159,10 @@ pub struct ConstraintSystem<F: FftField> {
     #[serde(skip)]
     pub fr_sponge_params: ArithmeticSpongeParams<F>,
 
+    /// Foreign field modulus parameter
+    #[serde_as(as = "Vec<o1_utils::serialization::SerdeAs>")]
+    pub foreign_modulus: Vec<F>,
```

Comment:
same as in verifier_index, I think both of the foreign stuff should be behind a single `Option`